src/hera/workflows/_runner/util.py

```diff
@@ -132,13 +132,21 @@ def _get_unannotated_type(key: str, f: Callable) -> Optional[type]:
     return type_
 
 
-def _is_str_kwarg_of(key: str, f: Callable) -> bool:
-    """Check if param `key` of function `f` has a type annotation of a subclass of str."""
+def _can_be_str_kwarg_of(key: str, f: Callable) -> bool:
```

Comment:
Could we change this name back? We're still only looking at types so `Optional[str]` still reads as a "str" type (vs a `Union[str, int, Other]`) , and it's more readable alongside the other functions.
```suggestion
def _is_str_kwarg_of(key: str, f: Callable) -> bool:
```

Reply:
Makes sense. fixed!